lib/processor/src/girouette/processor.clj

```diff
@@ -157,7 +167,15 @@
         (pp/pprint @file-data file-writer))
       (let [all-css-classes (into #{} (mapcat :css-classes) (vals @file-data))
             predef-garden (if preflight? preflight [])
-            all-garden-defs (into predef-garden (keep (:garden-fn @config)) (sort all-css-classes))]
+            all-garden-defs (into predef-garden (keep (:garden-fn @config)) (sort all-css-classes))
+            all-garden-defs (reduce (fn [garden [cls classes]]
+                                      (into garden
+                                            (util/apply-class-rules
+                                              garden
```

Comment:
The process is in `n^2`, it will be too slow on larger sizes of inputs.

Reply:
I think I understand now the use of the `keep` in your code, it was to provide all the garden rules already processed by Girouette. I prefer to assume that all the garden rules passed to the apply-class-rules are used, and change the way the process is implemented.